bundle/config/mutator/process_target_mode.go

```diff
@@ -87,6 +87,12 @@ func transformDevelopmentMode(b *bundle.Bundle) error {
 		// (model serving doesn't yet support tags)
 	}
 
+	for i := range r.RegisteredModels {
+		prefix = "dev_" + b.Config.Workspace.CurrentUser.ShortName + "_"
+		r.RegisteredModels[i].Name = prefix + r.RegisteredModels[i].Name
+		// (registered models in Unity Catalog don't yet support tags)
+	}
```

Comment:
Thought: these are UC resources, where I expect isolation at the catalog/schema level instead of by name.

If we rewrite this here, we cannot change it later because it triggers the recreation of the underlying models.

Also, to use these downstream, users must refer to `${resources.registered_models.my_model.name}` as an e.g. job parameter to make it work. Implying the name is no longer possible.

cc @lennartkats-db

Reply:
Good point, removed `registered_models` from getting modified by `process_target_mode`.